rclcpp/doc/proposed_node_parameter_callbacks.md

```diff
@@ -0,0 +1,33 @@
+# Proposed node parameters callback Design
+
+## Introduction:
+
+The original requirement came in **gazeb_ros_pkgs** for setting individual wheel slip params based on global wheel slip value [link to original issue](https://github.com/ros-simulation/gazebo_ros_pkgs/pull/1365). 
+The main idea was to set some parameters using `set_parameter` API once a given param was set successfully. 
+The requirement let to the discussion of supporting registered callbacks once the parameters are set and have been requested by users before.
```

Comment:
```suggestion
Additionally, it would be nice if users could be notified locally (via a callback) when parameters have been set successfully (i.e. post validation).
```